package_modules/yum

```diff
@@ -0,0 +1,190 @@
+#!/usr/bin/python
+
+import sys
+import os
+import subprocess
+import re
+
+
+rpm_cmd = os.environ.get('CFENGINE_TEST_RPM_CMD', "/bin/rpm")
+rpm_output_format = "Name=%{name}\nVersion=%{version}\nArchitecture=%{arch}\n"
+
+yum_cmd = os.environ.get('CFENGINE_TEST_YUM_CMD', "/usr/bin/yum")
+yum_options = "--quiet"
+
+def get_package_data():
+    pkg_string = ""
+    for line in sys.stdin:
+        if line.startswith("File="):
+            pkg_string = line.split("=", 1)[1].rstrip()
+            break
+
+    if not pkg_string:
+        return 1
```

Comment:
```
files = (line.split('=', 1)[1].rstrip() for line in sys.stdin if line.startswith('File='))
try: pkg_string = files.next()
except StopIteration: return 1
```
does all of the above.  Note that files is a raw generator (and you'd get a SyntaxError if you missed off the parentheses round it).  When we call its `.next()` we do the first iteration of it, which may iterate many lines of sys.stdin, but it'll stop on the first that starts with 'File=' and give you the rest of the line.  It won't bother with the rest of the iteration and it won't consume the rest of sys.stdin; no need for a break, just don't ask for any remaining values.

The iterator protocol, as used by `for` loops and other things that iterate over a container (or generator) goes: `seq = iter(container)` where `iter` is a builtin and any generator's `iter()` is itself; then it uses `seq.next()` as the value for the loop variable at each iteration, until calling `seq.next()` raises `StopIteration` at which point it gracefully exits the loop.  Any other exception from `.next()` is delivered up the stack as usual.